samples/nrf9160/aws_fota/src/main.c

```diff
@@ -39,7 +39,11 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT),
 /* Define the length of the IMEI AT COMMAND response buffer */
 #define CGSN_RESP_LEN 19
 #define IMEI_LEN 15
-#define CLIENT_ID_LEN (IMEI_LEN + sizeof("nrf-"))
+#if !defined(CONFIG_USE_NRF_CLOUD)
+#define CLIENT_ID_LEN (sizeof(CONFIG_CLOUD_CLIENT_ID_PREFIX) - 1 + IMEI_LEN)
+#else
+#define CLIENT_ID_LEN (sizeof(CONFIG_NRF_CLOUD_CLIENT_ID_PREFIX) - 1 + IMEI_LEN)
```

Comment:
```
main.c:45:31: error: 'CONFIG_NRF_CLOUD_CLIENT_ID_PREFIX' undeclared here (not in a function)
   45 | #define CLIENT_ID_LEN (sizeof(CONFIG_NRF_CLOUD_CLIENT_ID_PREFIX) - 1 + IMEI_LEN)

Reply:
Ah, I see, this happens if `USE_NRF_CLOUD` is `y` but `NRF_CLOUD` is `n`.

Reply:
... in a case where an app only wants nRF Connect for Cloud FOTA

Reply:
Yes I'll make a PR to your branch with some fixes.

Reply:
OK.

Could this be solved by defining the config, if `NRF_CLOUD=n` in the `aws_fota` Kconfig?

```c
if !NRF_CLOUD

config NRF_CLOUD_CLIENT_ID_PREFIX
	string "Prefix used when constructing the MQTT client ID from the IMEI"
	default "nrf-"

endif # !NRF_CLOUD
```

Reply:
yes in the sample config. as suggested in the PR.

Reply:
```
  391 |  snprintf(id_buf, len, "%s%s", CONFIG_CLOUD_CLIENT_ID_PREFIX, imei_buf);
      |                        ^~~~~~                                 ~~~~~~~~
../src/main.c:391:27: note: format string is defined here
  391 |  snprintf(id_buf, len, "%s%s", CONFIG_CLOUD_CLIENT_ID_PREFIX, imei_buf);
      |                           ^~
In file included from /opt/tools/zephyr-sdk/arm-zephyr-eabi/arm-zephyr-eabi/sys-include/stdio.h:800,
                 from ../src/main.c:8:
../src/main.c:391:2: note: '__builtin_snprintf' output between 5 and 23 bytes into a destination of size 20
  391 |  snprintf(id_buf, len, "%s%s", CONFIG_CLOUD_CLIENT_ID_PREFIX, imei_buf);
      |  ^~~~~~~~`

still some issues with these lines.

Reply:
Yeah, because this buffer is longer: https://github.com/nrfconnect/sdk-nrf/blob/208302083ec5591fad6ede4a46ecff36db8c52a5/samples/nrf9160/aws_fota/src/main.c#L378

Reply:
Changed to use `IMEI_LEN + 1` instead.